### PR TITLE
Generate: shorter XLA contrastive search tests

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

Makes XLA contrastive search tests shorter (in terms of tokens), to avoid flaky tests.

This is due to our recently failing CI for some models. The XLA code path passes the test with XLA compilation off -- i.e. the XLA code path returns the same as the non-XLA code path. However, with XLA compilation on, there is a chance of obtaining different results. I couldn't pinpoint the issue, but there is a possible explanation.

This may be due to numerical stability issues: contrastive search takes the maximum of a cosine distance between two hidden states [built from randomly initialized weights] as a penalty to the logits, which combined with the logits' low values [because the test model was untrained] could explain the mismatch.

👉 In any case, I was already planning on reinforcing contrastive search XLA testing with real examples on key models, like T5 and OPT.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks!

**Review by ydshieh — APPROVED** [review]
Thanks for taking care the issue :-)

**Comment by ydshieh** [comment]
And as always, if a one-line short comment could keep your explanation more visible for other (HF) developers, don't hesitate.
